nexus/src/app/mod.rs

```diff
@@ -915,12 +917,43 @@ impl Nexus {
 
     /// Reliable persistent workflows can request that sagas be executed by
     /// sending a SagaRequest to a supplied channel. Execute those here.
-    pub(crate) async fn handle_saga_request(&self, saga_request: SagaRequest) {
+    pub(crate) async fn handle_saga_request(
```

Comment:
I see that this channel was here already, but I find it to be a poor workaround for for the full API provided by the `Nexus` type. For one thing, not all the methods on sagas are available. For another, you can actually see what errors are reported directly when operating on the `sec_client` to start a saga. The `sec_client` used by nexus is fully clonable, and I feel like a better option would be to extract the Nexus API into a type wrapper around the `sec_client` and then allow access to that from `Nexus`. 

I only bring this up because I'm working on saga migration, and just ran into this same problem and wanted to see how you dealt with it here. Dealing with everything being a method on `Nexus` is getting super annoying, and I'd love to start breaking it up. Maybe that does mean just extracting the type and allowing callers of the nexus app layer to use it from nexus rather than providing methods on nexus. I think this can be done easily enough in a separate PR, and I'm happy to take a stab at it, although I'll be out a bunch tomorrow.

What do you think?

Reply:
Closing this comment since we've long since sorted all this out and have a much nicer interface for starting sagas now :)